test/test_sparse_csr.py

```diff
@@ -835,6 +833,52 @@ def test_to_dtype(self, layout, device, dtype):
                 dense_to_dtype = sparse.to_dense().to(to_dtype)
                 self.assertEqual(sparse_to_dtype.to_dense(), dense_to_dtype)
 
+    @all_sparse_compressed_layouts()
+    @parametrize("index_dtype", [torch.int32, torch.int64])
+    @dtypes(*all_types_and_complex_and(torch.half, torch.bfloat16, torch.bool))
+    def test_select_copy(self, device, dtype, index_dtype, layout):
+
+        def is_view_of(base, other):
+            # a shameless copy of TestViewOps.is_view_of
+            if ((not other._is_view() or
+                 other is base or
+                 other._base is not base or
+                 base.device != other.device)):
+                return False
+            if base.device.type == 'cpu' or base.device.type == 'cuda':
+                if base._storage().data_ptr() != other._storage().data_ptr():
```

Comment:
nit: hmm, do we have `device.type`s which are different from `cpu/cuda`?

Reply:
The explanation is in
https://github.com/pytorch/pytorch/blob/1f7c0ff6e799e7bde94975f7a5bbec39a69ab8f6/test/test_view_ops.py#L102-L103